### PR TITLE
chore: adds a new option only-acir

### LINKED ISSUE #3119: Add compile options so that nargo can output compressed acir and compressed witnesses
### Problem

For a user who wants to use nargo to compile his circuit and then output artifacts that can be fed straight into a backend, they need to use jq to extract the fields from the json file.

### Happy Case

We allow users to do `nargo compile --only-acir` or some flag to specify that you only want to output the acir 

### Alternatives Considered

_No response_

### Additional Context

_No response_

### Would you like to submit a PR for this Issue?

No

### Support Needs

_No response_

### PR DESCRIPTION
**Pull request by guipublic**
# Description

## Problem\*

Resolves #3119

## Summary\*

Adds a new option only-acir which compiles acir bytecode into acir.gz
generates witness as .gz instead of .tr
use only-acir when generating the artifacts and remove the un-needed steps from rebuild.sh

## Documentation\*

Check one:
- [ ] No documentation needed.
- [X] Documentation included in this PR.
- [ ] **[Exceptional Case]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [X] I have tested the changes locally.
- [X] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: tooling/nargo_cli/src/cli/fs/program.rs [hunk 1]
```diff
@@ -18,6 +19,19 @@ pub(crate) fn save_program_to_file<P: AsRef<Path>>(
     save_build_artifact_to_file(compiled_program, &circuit_name, circuit_dir)
 }
 
+/// Writes the bytecode as acir.gz (without uuencoding)
```
Comment (kevaundray) [inline tooling/nargo_cli/src/cli/fs/program.rs:22]: typo: uuencoding

File: compiler/noirc_driver/src/lib.rs [hunk 2]
```diff
@@ -61,6 +61,9 @@ pub struct CompileOptions {
     /// Suppress warnings
     #[arg(long, conflicts_with = "deny_warnings")]
     pub silence_warnings: bool,
+
```
Comment (kevaundray) [inline compiler/noirc_driver/src/lib.rs:64]: can you add a doc comment here?

Comment (guipublic) [reply compiler/noirc_driver/src/lib.rs:64]: Done

File: tooling/nargo_cli/src/cli/fs/program.rs (hunk 1 above)
Comment (kevaundray) [inline tooling/nargo_cli/src/cli/fs/program.rs:22]:
```suggestion
/// Writes the bytecode as acir.gz
```

**Review by kevaundray — APPROVED** [review]

**Comment by kevaundray** [comment]
Just need to resolve conflicts and I think we don't need to add this to the docs (at least not initially) as we can make the flag be hidden in the CLI

**Comment by guipublic** [comment]
I resolved the conflicts, hid the flag and did not add back the documentation.

File: docs/docs/reference/nargo_commands.md [hunk 3]
```diff
@@ -99,14 +99,15 @@ You can also use "build" as an alias for compile (e.g. `nargo build`).
...
-| `-h, --help`          | Print help                                                   |
+| Option                | Description                                                                 |
+| --------------------- | ----------------------------------------------------------------------------|
+| `--package <PACKAGE>` | The name of the package to compile                                          |
+| `--workspace`         | Compile all packages in the workspace                                       |
+| `--print-acir`        | Display the ACIR for compiled circuit                                       |
+| `--deny-warnings`     | Treat all warnings as errors                                                |
+| `--silence-warnings`  | Suppress warnings                                                           |
+| `--only-acir`         | generates ACIR representation into acir.gz, instead of JSON build artifact  |
+| `-h, --help`          | Print help                                                                  |
```
Comment (kevaundray) [inline docs/docs/reference/nargo_commands.md:110]:
```suggestion
| `-h, --help`          | Print help                                                                  |
```

File: docs/docs/reference/nargo_commands.md [hunk 4]
```diff
@@ -99,14 +99,14 @@ You can also use "build" as an alias for compile (e.g. `nargo build`).
...
-| Option                | Description                                                  |
-| --------------------- | ------------------------------------------------------------ |
-| `--package <PACKAGE>` | The name of the package to compile                           |
-| `--workspace`         | Compile all packages in the workspace                        |
-| `--print-acir`        | Display the ACIR for compiled circuit                        |
-| `--deny-warnings`     | Treat all warnings as errors                                 |
-| `--silence-warnings`  | Suppress warnings                                            |
-| `-h, --help`          | Print help                                                   |
+| Option                | Description                                                                 |
+| --------------------- | ----------------------------------------------------------------------------|
```
Comment (kevaundray) [inline docs/docs/reference/nargo_commands.md:103]:
```suggestion
| --------------------- | ------------------------------------------------------------ |
```

File: docs/docs/reference/nargo_commands.md (hunk 3 above)
Comment (Savio-Sou) [reply docs/docs/reference/nargo_commands.md]: Why did we remove this?

**Pull request merged**